### PR TITLE
Add new TxFilter

### PR DESCRIPTION
**Pull request by MansurovB-source**
## Changes:
Added new filter

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 1]
```diff
@@ -110,13 +110,16 @@ public partial class TxPool : ITxPool, IDisposable
             _filterPipeline.Add(new TooFarNonceFilter(txPoolConfig, _accounts, _transactions, _logger));
             _filterPipeline.Add(new TooExpensiveTxFilter(_headInfo, _accounts, _logger));
             _filterPipeline.Add(new FeeToLowFilter(_headInfo, _accounts, _transactions, _logger));
+            _filterPipeline.Add(new NotEnoughBalanceFilter(_headInfo, _accounts, _logger));
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:113]: How does this filter differ from TooExpensiveTxFilter?

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
Looks redundant to me

File: src/Nethermind/Nethermind.TxPool/AddTxResult.cs [hunk 2]
```diff
@@ -25,52 +25,52 @@ public enum AddTxResult
...
         /// </summary>
         Filtered,
-        
+
         /// <summary>
         /// Transaction gas limit exceeds the block gas limit.
         /// </summary>
         GasLimitExceeded,
-        
+
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.TxPool/AddTxResult.cs:58]: Minor: could you try to change your IDE settings?

File: src/Nethermind/Nethermind.TxPool/TxPool.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.TxPool/TxPool.cs:113]: We will discuss it with @MansurovB-source and @marcindsobczak

**Pull request closed without merging**